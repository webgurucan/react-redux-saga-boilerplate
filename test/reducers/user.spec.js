import expect from 'expect';
import { REHYDRATE } from 'redux-persist/constants';

import reducers from 'reducers';
import { userState } from 'reducers/user';
import { ActionTypes } from 'constants/index';

describe('User', () => {
  it('should return the initial state', () => {
    expect(reducers.user(undefined, {}))
      .toEqual(userState);
  });

  it(`should handle ${REHYDRATE} with payload`, () => {
    expect(reducers.user(undefined, { type: REHYDRATE, payload: { user: [] } }))
      .toEqual({ ...userState, rehydrated: true });
  });

  it(`should handle ${REHYDRATE} without payload`, () => {
    expect(reducers.user(undefined, { type: REHYDRATE, payload: { } }))
      .toEqual(userState);
  });

  it(`should handle ${ActionTypes.USER_LOGIN_SUCCESS}`, () => {
    expect(reducers.user(undefined, { type: ActionTypes.USER_LOGIN_SUCCESS }).logged)
      .toEqual(true);
  });

  it(`should handle ${ActionTypes.USER_LOGOUT_SUCCESS}`, () => {
    expect(reducers.user(undefined, { type: ActionTypes.USER_LOGOUT_SUCCESS }).logged)
      .toEqual(false);
  });
});
