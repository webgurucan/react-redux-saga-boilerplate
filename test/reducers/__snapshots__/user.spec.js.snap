// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should handle USER_LOGIN_REQUEST 1`] = `
Object {
  "isAuthenticated": false,
  "isRunning": true,
}
`;

exports[`User should handle USER_LOGIN_SUCCESS 1`] = `
Object {
  "isAuthenticated": true,
  "isRunning": false,
}
`;

exports[`User should handle USER_LOGOUT_REQUEST 1`] = `
Object {
  "isAuthenticated": false,
  "isRunning": true,
}
`;

exports[`User should handle USER_LOGOUT_SUCCESS 1`] = `
Object {
  "isAuthenticated": false,
  "isRunning": false,
}
`;

exports[`User should return the initial state 1`] = `
Object {
  "isAuthenticated": false,
  "isRunning": false,
}
`;
